### PR TITLE
feat: add linter practice for C#

### PR DESCRIPTION
**Pull request by gagandeepp**
## Description

Add linter practice for c# project. Checking for the following packages:

golang.org/x/lint
github.com/golangci/golangci-lint

## Motivation and Context

resolves #451 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have added new practice to practice list in README.md.
- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by gagandeepp** [comment]
Which Linter you want to use ??

**Comment by adelkahomolova** [comment]
Hi, firstly I'm really sorry for the long delay. I had a crazy week. @gagandeepp 
I don't know which linter you should check. I'm not a C# developer and choosing the right one is a part of the task. 
I'll also ask you for a renaming pull request since it's not an initial commit and also for fill in the PR template. Thanks :)

File: src/practices/C#/CsharpLinterUsedPractice.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  url: 'https://dxkb.io/p/linting',
+})
+export class CSharpLinterUsedPractice implements IPractice {
+  async isApplicable(ctx: PracticeContext): Promise<boolean> {
+    return ctx.projectComponent.language === ProgrammingLanguage.CSharp;
+  }
+
+  async evaluate(ctx: PracticeContext): Promise<PracticeEvaluationResult> {
+    if (ctx.packageInspector) {
+      if (ctx.packageInspector.hasOneOfPackages(['golang.org/x/lint', 'github.com/golangci/golangci-lint'])) {
```
Comment (adelkahomolova) [inline src/practices/C#/CsharpLinterUsedPractice.ts:21]: Choose the most used C# linter.

**Review by adelkahomolova — CHANGES_REQUESTED** [review]

**Comment by adelkahomolova** [comment]
Hi, @gagandeepp , how are you? Is there any progress with this PR? :)

**Comment by gagandeepp** [comment]
> Hi, @gagandeepp , how are you? Is there any progress with this PR? :)

In C# we used IDE built in code analysis which is available as DLL ,so I am not sure whether this change can be implemented.

**Comment by adelkahomolova** [comment]
@gagandeepp Look at this article. Some of them should be able to use for this practice. https://medium.com/@michaelparkerdev/linting-c-in-2019-stylecop-sonar-resharper-and-roslyn-73e88af57ebd

**Comment by adelkahomolova** [comment]
Hi, @gagandeepp , is there any progress? :)

**Comment by gagandeepp** [comment]
> Hi, @gagandeepp , is there any progress? :)

Sorry I am unable to a make progress on this

**Pull request closed without merging**